code/modules/mob/inventory.dm

```diff
@@ -181,8 +181,17 @@ var/list/slot_equipment_priority = list( \
 			qdel(grab)
 			. = TRUE
 		return
+	if(!is_space_available(get_active_hand(), Target ? get_turf(Target) : get_turf(loc)))
+		return
 	return hand ? drop_l_hand(Target) : drop_r_hand(Target)
 
+/mob/proc/is_space_available(var/obj/item/W, var/atom/target)
```

Comment:
Wouldn't it make more sense to define this on the atom itself rather than only when a mob is involved?

Reply:
> I definitely feel like this is way too high level as a solution for the problem you described on Discord. There are a ton of edge cases and ways for atoms to pile up on turfs or inside other atoms beyond when dropped by a mob. You could consider overriding `/atom/Enter()` to fail if the contents bound is exceeded, for example, and handle that gracefully in `dropInto()` to avoid breaking mob drop/unequip calls.

Counterpoint: I think handling all edge cases would be a massive undertaking, and a brittle solution. The effort required to handle all edge cases would exceed the value of the solution itself as well. Handling just the happy-path is efficient enough. That's my cost-benefit analysis.

I've looked at a few exceptions, things like mobs or containers getting Destroyed can immediately drop all contents into their occupying turf. Same for deconstruction and the like. In this way you can still bypass the atom/turf content limit, but it is *much* harder than it was before. I think that fits my design goals.

Reply:
The only time it's called is when a mob is involved.